markers/sauce.py

```diff
@@ -14,7 +17,31 @@ def pytest_addoption(parser):
 def pytest_configure(config):
     config.addinivalue_line('markers', __doc__.splitlines()[0])
     if config.option.sauce:
+        conf.env['browser']['sauce'] = True
         if config.option.markexpr:
             config.option.markexpr = 'sauce and ({})'.format(config.option.markexpr)
         else:
             config.option.markexpr = 'sauce'
+
+
+def pytest_runtest_setup(item):
+    browser_name = None
+    if 'chrome' in str(item).lower():
+        browser_name = 'chrome'
+    elif 'firefox' in str(item).lower():
+        browser_name = 'firefox'
+    browser_options = conf.env.get('browser', {})['webdriver_options']
+    if 'desired_capabilities' not in browser_options:
+        browser_options['desired_capabilities'] = {}
+    if item:
+        test_name = re.findall(r"'(test.*)'", str(item))[0]
+        browser_options['desired_capabilities']['name'] = test_name
+    if browser_name:
+        browser_options['desired_capabilities']['browserName'] = browser_name
+    else:
+        browser_options['desired_capabilities']['browserName'] = 'firefox'
+    browser.ensure_browser_open(browser_name=browser_name)
+
+
+def pytest_runtest_teardown():
+    browser.quit()
```

Comment:
Can we wrap this with a check to make sure the sauce config option has been requested